lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -87,28 +87,29 @@ static FIRRTLType getFIRRTLType(Type type) {
               false, getFIRRTLType(it.value())));
         return BundleType::get(elements, context);
       })
-      .Default([&](Type) { return FIRRTLType(); });
+      .Default([&](Type) { return FIRRTLBaseType(); });
 }
 
 /// Creates a new FIRRTL bundle type based on an array of port infos.
-static FIRRTLType portInfosToBundleType(MLIRContext *ctx,
-                                        ArrayRef<PortInfo> ports) {
+static FIRRTLBaseType portInfosToBundleType(MLIRContext *ctx,
+                                            ArrayRef<PortInfo> ports) {
   using BundleElement = BundleType::BundleElement;
   llvm::SmallVector<BundleElement, 4> elements;
   for (auto &port : ports) {
-    elements.push_back(
-        BundleElement(port.name, port.direction == Direction::Out, port.type));
+    elements.push_back(BundleElement(port.name,
+                                     port.direction == Direction::Out,
+                                     port.type.cast<FIRRTLBaseType>()));
   }
   return BundleType::get(elements, ctx);
 }
 
 /// Return a FIRRTL bundle type (with data, valid, and ready subfields) given a
 /// standard data type.
-static FIRRTLType getBundleType(Type type) {
+static FIRRTLBaseType getBundleType(Type type) {
   // If the input is already converted to a bundle type elsewhere, itself will
   // be returned after cast.
-  if (auto firrtlType = type.dyn_cast<FIRRTLType>())
```

Comment:
Was this supposed to be `auto firrtlType = type.dyn_cast<BundleType>()`

Reply:
Not sure, both make sense to me for different reasons.  I'll switch to BundleType :).